File: src/main/osu/beatmap/ComboColor.java
package main.osu.beatmap;

import main.osu.RGBColor;

/**
 * An extension to <code>RGBColor</code> specifically for combo colors.
 */
public class ComboColor extends RGBColor {
    /**
     * The index of this combo color.
     */
    public final int index;

    public ComboColor(int index, RGBColor color) {
        super(color);

        this.index = index;
    }
}

File: src/com/rian/osu/beatmap/ComboColor.kt
package com.rian.osu.beatmap

import com.reco1l.framework.*

/**
 * A wrapper of [Color4] specifically for combo colors.
 */
data class ComboColor(
    /**
     * The index of this combo color.
     */
    @JvmField val index: Int,

    /**
     * The wrapped [Color4].
     */
    val color: Color4
)
